documents/toolbox-population.md

```diff
@@ -0,0 +1,80 @@
+
+### Limitations
+
+* This approach only works reliably for WPF .NET Framework assemblies. The controls in registered assemblies are enumerated via .NET Reflection which does not work reliably on .NET Core assemblies within a .NET Framework process such as Visual Studio.
+* Since registration requires modifying the registry, you need to provide an installer of some sort.
+
+## Toolbox items from referenced NuGet packages (Visual Studio 15.0 and later)
+
+If a NuGet package referenced by a XAML project contains a [tools\VisualStudioToolsManifest.xml file](https://docs.microsoft.com/en-us/nuget/guides/create-ui-controls), Toolbox will show the Toolbox items listed in the manifest.
+
+You can try this by adding a reference to your own package or to [our WPF .NET Core sample package](#how-to-build-our-wpf-net-core-sample-package).
+
+### Target Framework-specific Manifests
+
+In Visual Studio 16.6 and later, Toolbox population supports multiple VisualStudioToolsManifest.xml files per package -- the maniest in the tools root plus additional manifests for specific Target Framework Monikers (TFMs) in subdirectories of tools. The Toolbox will show the items from the manifest that best matches the target framework of the current project, falling back to the manifest in the tools root if there is no better match. The manifest in the tools root is also required for compatibility with older versions of Visual Studio.
+
+Here's an example:
+
+| File path                                         | Will be used for projects targetting...                     |
+| :------------------------------------------------ | :---------------------------------------------------------- |
+| \tools\net47\VisualStudioToolsManifest.xml        | .NET Framework >= 4.7                                       |
+| \tools\netcoreapp31\VisualStudioToolsManifest.xml | .NET Core >= 3.1                                            |
+| \tools\VisualStudioToolsManifest.xml              | .NET Framework < 4.7, .NET Core < 3.1, and other frameworks |
+
+## Toolbox items from unreferenced NuGet packages (Visual Studio 16.7 Preview 2 and later)
+
+If a NuGet package in a NuGet fallback folder contains a [tools\VisualStudioToolsManifest.xml file](https://docs.microsoft.com/en-us/nuget/guides/create-ui-controls), Toolbox will show the Toolbox items listed in the manifest for any project that is compatible with the package, even projects that do not reference that package.
+
+### Limitations of the fallback folder approach
+
+* It currently only supports WPF .NET Core packages and projects.
+* It only supports one manifest per package, in the root of the tools directory. [Manifests for specific Target Framework Monikers](#target-framework-specific-manifests) are not supported.
```

Comment:
I thought the manifest limitation is also only temporary for Preview 2.

Reply:
It's temporary, but I'm not planning to fix it until 16.8, or whenever we add support for platforms beyond .NETCoreApp 3.x. I didn't want to commit to a timeframe for resolving these issues.

Reply:
Perhaps we could say then that it "currently only supports..."

---
In reply to: [433435462](https://github.com/microsoft/xaml-designer-extensibility/pull/14#discussion_r433435462) [](ancestors = 433435462)

Reply:
Good point. Fixed.